int32 rgb
float32 timeForRotation
float32 totalDuration
